elytron/WFLY-16306-keycloak-saml-feature-pack.adoc

```diff
@@ -0,0 +1,330 @@
+
+In particular, when the `keycloak-saml` layer is provisioned, the following steps will occur:
+
+ # Install a set of modules under `modules/system/add-ons`
+ # Register the `org.keycloak.keycloak-saml-adapter-subsystem` extension.
+ # Add an empty `urn:jboss:domain:keycloak:1.4` subsystem.
+
+The `keycloak-client-saml` layer will depend on the `keycloak-saml` layer and when provisioned, the following additional steps will occur:
+
+ # Add the required resources to the `elytron` subsystem to result in an `http-authentication-factory` supporting SAML being available.
+ # Update the `application-security-domain` mapping in the `undertow` subsystem to map the default security domain to the new `http-authentication-factory`.
+
+The CLI scripts that would have been added via the manual installation steps won't be added to the resulting installation
+since the required resources will be added already during provisioning.
+
+Note that Keycloak and WildFly have different release cadences. This means anyone using Galleon to keep their WildFly
+installation up to date should be able to receive updates to the core WildFly installation independently of the new
+Keycloak feature pack. Having the new Keycloak feature pack installed should not prevent the server from being updated
+once later WildFly versions are available. Similarly, WildFly releases should not trigger the need for an update to Keycloak.
+
+==== Bare Metal vs. Cloud Use Cases
+
+For the bare metal use case, both layers (`keycloak-saml` and `keycloak-client-saml`) will be needed. For the cloud use case,
+only the `keycloak-saml` layer is needed. The rest of the `elytron` and `undertow` subsystem configuration will be handled
+by a launch script at startup that generates CLI scripts.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Published artifacts will be available from online maven repositories only. No offline maven repository support will be provided.
+
+The feature pack and resulting layer being provisioned are for securing web applications accessed directly via HTTP.
+This feature request does not extend to securing management interfaces, Remoting invocations, or EJB over HTTP invocations.
+
+== Examples
+
+This section contains examples of how the new feature pack will be installed using different approaches.
+
+=== Provisioned using the Galleon CLI
+
+A server installation can be provisioned using the Galleon CLI which would need to be downloaded and installed first.
+
+Start the CLI:
+
+----
+java -jar galleon-cli-5.0.2.Final-SNAPSHOT.jar
+----
+
+First provision a WildFly server with the `web-server` later:
+
+----
+install wildfly:current --layers=web-server --dir=/home/fjuma/tmp/galleon/wildfly
+----
+
+Then install the Keycloak Elytron SAML adapter:
+
+----
+install org.keycloak:keycloak-saml-adapter-galleon-pack:18.0.0 --layers=keycloak-client-saml --dir=/home/fjuma/tmp/galleon/wildfly
+----
+
+=== Provisioned using org.jboss.galleon:galleon-maven-plugin.
```

Comment:
Although galleon maven plugin usage is correct, the maven plugin that we should cover is the org.wildfly.plugins:wildfly-maven-plugin (starting version 3) that allows to provision a server, apply CLI scripts, ... Similar to the Bootable JAR experience. You can see doc there: https://docs.wildfly.org/wildfly-maven-plugin/package-example.html